### PR TITLE
ipvs: fix string check for IPVS protocol during graceful termination

### PR DESCRIPTION
**Pull request by andrewsykim**
Signed-off-by: Andrew Sy Kim <kiman@vmware.com>

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
In https://github.com/kubernetes/kubernetes/pull/77802 we removed graceful termination for UDP (causes many issues when there are too many connections/real servers). It turns out that in `pkg/util/ipvs` we actually convert the protocol values from netlink to upper case strings ("TCP", "UDP", etc, see [protocolToString](https://github.com/kubernetes/kubernetes/blob/master/pkg/util/ipvs/ipvs_linux.go#L299-L309)) even though the string passed into `docker/libnetwork/ipvs` is lower case string (see [stringToProtocol](https://github.com/kubernetes/kubernetes/blob/master/pkg/util/ipvs/ipvs_linux.go#L285-L296)). 

This PR fixes this bug by always checking upper case values. As a follow-up we should consolidate the string casing for utils/ipvs to ensure this doesn't happen again. The PR for this will be larger so wanted to get the bug fix only change in before we cut v1.15 and for easier cherry-picking for previous versions. 

I'm missing context as to why we use lower case values for `stringToProtocol` and upper case values for `protocolToString`, would appreciate some background there @m1093782566  @lbernail. 

**Which issue(s) this PR fixes**:
Fixes #https://github.com/kubernetes/kubernetes/issues/78993

DNS service in the reported issue went down due to too many stale real server entries being held by graceful termination. 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
Fix a string comparison bug in IPVS graceful termination where UDP real servers are not deleted.
```

### CODE REVIEW

**Comment by andrewsykim** [comment]
More proper fix for a follow-up PR would be to change the string values returned in [protocolToString](https://github.com/kubernetes/kubernetes/blob/master/pkg/util/ipvs/ipvs_linux.go#L299-L309) to lower case values to be consistent with values passed into `libnetwork/ipvs`. Didn't do that in this PR since I'm not sure what the implications are yet.

**Comment by andrewsykim** [comment]
Also created https://github.com/kubernetes/kubernetes/issues/79001 re: e2e coverage for IPVS graceful termination

**Comment by lbernail** [comment]
Nice catch
I tested with "!=tcp" before we decided to disable it only for UDP. So this worked (but would have broken TCP graceful termination...)

I agree that we should standardize on lower or upper everywhere to avoid this type of issue

**Comment by lbernail** [comment]
We'll need to cherry-pick this in 1.14, 1.13 and 1.12

**Comment by bazzargh** [comment]
(for the future cleanup, not this patch) Would suggest moving those strings into constants for consistency.

**Comment by liggitt** [comment]
needs approval. once approved, go ahead and open the cherrypick against 1.15, since the master merge pool is full

**Comment by tpepper** [comment]
@kubernetes/patch-release-team

**Comment by liggitt** [comment]
> @kubernetes/patch-release-team

note that this is targeting v1.15.0

**Review by khenidak — APPROVED** [review]

**Comment by andrewsykim** [comment]
Created https://github.com/kubernetes/kubernetes/pull/79044 for faster turn over into 1.15 if/when this one gets approved (please approve this one first).

**Comment by thockin** [comment]
Thanks!

/lgtm
/approve

**Comment by andrewsykim** [comment]
@thockin can you approve the cherry-pick PR as well https://github.com/kubernetes/kubernetes/pull/79044, thank you

**Pull request merged**
